src/pull_request.ts

```diff
@@ -53,19 +47,76 @@ export async function handlePullRequest() {
   let overviewComment = existingComments.find((comment) =>
     comment.body?.includes(OVERVIEW_MESSAGE_SIGNATURE)
   );
+  const isIncrementalReview = !!overviewComment;
+
+  // Maybe fetch review comments
+  const reviewComments = isIncrementalReview
+    ? (
+        await octokit.rest.pulls.listReviewComments({
+          ...context.repo,
+          pull_number: pull_request.number,
+        })
+      ).data
+    : [];
+
+  // Get modified files
+  const { data: files } = await octokit.rest.pulls.listFiles({
+    ...context.repo,
+    pull_number: pull_request.number,
+  });
+  let filesToReview = files.map((file) => parseFileDiff(file, reviewComments));
+  info(`successfully fetched file diffs`);
+
+  let commitsReviewed: string[] = [];
   if (overviewComment) {
+    info(`running incremental review`);
+    try {
+      const payload = JSON.parse(
+        overviewComment.body
+          ?.split(PAYLOAD_TAG_OPEN)[1]
+          .split(PAYLOAD_TAG_CLOSE)[0] || "{}"
+      );
+      commitsReviewed = payload.commits;
+    } catch (error) {
+      warning(`error parsing overview payload: ${error}`);
+    }
+
+    // Check if there are any incremental changes
+    const lastCommitReviewed = commitsReviewed.length
+      ? commitsReviewed[commitsReviewed.length - 1]
+      : null;
```

Comment:
Consider using the more modern `at()` method with optional chaining to simplify the code and make it more readable:
```typescript
const lastCommitReviewed = commitsReviewed.at(-1) ?? null;
```
This approach is more concise and handles empty arrays gracefully.